src/core/managers/internal/LedgerConnectStateManager/LedgerConnectStateManager.ts

```diff
@@ -130,4 +184,127 @@ export class LedgerConnectStateManager<
+    if (!this.ledgerConnectElement) {
+      await this.createLedgerConnectElement();
+    }
+
+    if (!this.ledgerConnectElement) {
+      return null;
+    }
+
+    if (!this.eventBus) {
+      this.eventBus = await this.ledgerConnectElement.getEventBus();
+    }
+
+    if (!this.eventBus) {
+      throw new Error(ProviderErrorsEnum.eventBusError);
+    }
+
+    return this.eventBus;
+  }
+
+  public async createLedgerConnectElement(
+    anchor: HTMLElement | undefined = this.anchor
+  ): Promise<LedgerConnectPanel | null> {
+    if (this.ledgerConnectElement) {
+      return this.ledgerConnectElement;
+    }
+
+    if (!this.isCreatingElement) {
+      this.isCreatingElement = true;
+      const element = anchor
+        ? await createUIElement<LedgerConnect>({
+            name: UITagsEnum.LEDGER_CONNECT,
+            anchor
+          })
+        : await createUIElement<LedgerConnectPanel>({
+            name: UITagsEnum.LEDGER_CONNECT_PANEL
+          });
+
+      this.ledgerConnectElement = element || null;
+      await this.getEventBus();
+      this.isCreatingElement = false;
+    }
+
+    if (!this.ledgerConnectElement) {
+      throw new Error('Failed to create ledger connect element');
+    }
+
+    return this.ledgerConnectElement;
+  }
+
+  private async setupEventListeners() {
+    if (!this.ledgerConnectElement) {
+      await this.createLedgerConnectElement();
+    }
+
+    if (!this.eventBus) {
+      return;
+    }
+
+    this.eventBus.subscribe(
+      LedgerConnectEventsEnum.CLOSE_LEDGER_CONNECT_PANEL,
+      this.handleCloseLedgerConnect.bind(this)
+    );
+  }
+
+  private handleCloseLedgerConnect() {
+    this.isOpen = false;
+    this.resetData();
+  }
+
+  public subscribeToProviderInit(
+    onRetry: () => void,
+    onCancel: () => void
+  ): void {
+    if (!this.eventBus) {
+      return;
+    }
+
+    this.eventBus.subscribe(LedgerConnectEventsEnum.CONNECT_DEVICE, onRetry);
+    this.eventBus.subscribe(
+      LedgerConnectEventsEnum.CLOSE_LEDGER_CONNECT_PANEL,
+      onCancel
+    );
+  }
+
+  public unsubscribeFromProviderInit(
+    onRetry: () => void,
+    onCancel: () => void
+  ): void {
+    if (!this.eventBus) {
+      return;
+    }
+
+    this.eventBus.unsubscribe(LedgerConnectEventsEnum.CONNECT_DEVICE, onRetry);
+    this.eventBus.unsubscribe(
+      LedgerConnectEventsEnum.CLOSE_LEDGER_CONNECT_PANEL,
+      onCancel
+    );
+  }
+
+  public destroy() {
+    if (this.eventBus) {
+      this.eventBus.unsubscribe(
+        LedgerConnectEventsEnum.CLOSE_LEDGER_CONNECT_PANEL,
+        this.handleCloseLedgerConnect.bind(this)
+      );
+
+      this.eventBus = null;
+    }
+
+    if (this.ledgerConnectElement) {
+      const parentElement = this.ledgerConnectElement.parentElement;
+
+      if (parentElement) {
+        parentElement.removeChild(this.ledgerConnectElement);
+      }
+
+      this.ledgerConnectElement = null;
+    }
+
+    this.isOpen = false;
+  }
```

Comment:
Is this file not too long? It's a little hard to track, it's beyond 300 lines of code.